packages/core/src/stepper/useStepperReducer.ts

```diff
@@ -0,0 +1,27 @@
+import { useCallback, useReducer } from "react";
+import type { StepRecord } from "./Step";
+import {
+  StepperReducer,
+  type StepperReducerDispatch,
+  type StepperReducerState,
+} from "./internal/StepperReducer";
+import type { StepperReducerOptions } from "./internal/StepperReducer";
+import { initStepperReducerState } from "./internal/utils";
+
+/**
+ * @experimental
+ * This hook is experimental and may change in future releases.
+ */
+export function useStepperReducer(
```

Comment:
I'm inclined to make reducer code as part of pattern not component, but want to see how others think as well